test/unit/sqs/index.test.ts

```diff
@@ -22,140 +23,296 @@ describe("SqsQueue tests", () => {
 		handle: jest.fn()
 	};
 
+	const mockErrorHandler = {
+		handle: jest.fn()
+	};
+
 	const generatePayload = (): TestMessage => ({ msg: uuidv4() });
 
-	const createSqsQueue = () => {
+	const createSqsQueue = (timeout: number) => {
 		return new SqsQueue({
 			queueName: TEST_QUEUE_NAME,
 			queueUrl: TEST_QUEUE_URL,
 			queueRegion: TEST_QUEUE_REGION,
-			longPollingIntervalSec: 0
+			longPollingIntervalSec: 0,
+			timeoutSec: timeout,
+			maxAttempts: 3
 		},
-		mockRequestHandler);
+		mockRequestHandler,
+		mockErrorHandler);
 	};
 
 	let queue: SqsQueue<TestMessage>;
 
-	beforeEach(() => {
-		queue = createSqsQueue();
-		queue.start();
-	});
+	describe("Normal execution tests", () => {
 
-	afterEach(async () => {
-		queue.stop();
-		await queue.waitUntilListenerStopped();
-	});
+		beforeEach(() => {
+			queue = createSqsQueue(10);
+			queue.start();
+		});
 
-	test("Message gets received", (done: DoneCallback) => {
+		afterEach(async () => {
+			queue.stop();
+			await queue.waitUntilListenerStopped();
+		});
+
+		test("Message gets received", (done: DoneCallback) => {
 
-		const testPayload = generatePayload();
+			const testPayload = generatePayload();
 
-		mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
-			expect(context.payload).toStrictEqual(testPayload);
-			done();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
+				expect(context.payload).toStrictEqual(testPayload);
+				done();
+			});
+			queue.sendMessage(testPayload);
 		});
-		queue.sendMessage(testPayload);
-	});
 
 
-	test("Queue is restartable", async (done: DoneCallback) => {
+		test("Queue is restartable", async (done: DoneCallback) => {
 
-		const testPayload = generatePayload();
+			const testPayload = generatePayload();
 
-		mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
-			expect(context.payload).toStrictEqual(testPayload);
-			done();
+			mockRequestHandler.handle.mockImplementation((context: Context<TestMessage>) => {
+				expect(context.payload).toStrictEqual(testPayload);
+				done();
+			});
+
+			queue.stop();
+
+			//delaying to make sure all asynchronous invocations inside the queue will be finished and it will stop
+			await queue.waitUntilListenerStopped();
```

Comment:
Wouldn't make more sense to have the stop function be async to know when it's stopped instead of calling another function?